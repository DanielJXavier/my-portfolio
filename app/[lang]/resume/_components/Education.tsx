import { useParams } from "next/navigation";

import { getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import { education } from "@/[lang]/(pages)/education/_config";

import VisibilityControl from "./VisibiltyControl";

export default function Education() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    education: { items: educationStrings },
    resume: { sectionTitles },
  } = getDictionary(lang);

  return (
    <section>
      <h2 className="text-[16pt] font-bold">{sectionTitles.education}</h2>
      <ul className="flex flex-col gap-y-3">
        {education.map(({ schoolId, schoolName, endYear }, i) => (
          <VisibilityControl
            key={`education-item-${i}`}
            className="items-baseline break-inside-avoid"
          >
            <div>
              <p className="text-[12pt] font-bold">
                {educationStrings[schoolId].degree},{" "}
                {educationStrings[schoolId].fieldOfStudy} ({endYear})
              </p>
              <p className="text-[12pt]">{schoolName}</p>
            </div>
          </VisibilityControl>
        ))}
      </ul>
    </section>
  );
}
