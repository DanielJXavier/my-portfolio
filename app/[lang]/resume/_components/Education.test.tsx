import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import Education from "./Education";

import { useParams } from "next/navigation";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

describe("Education component", () => {
  it("renders the component", () => {
    const { container } = render(<Education />);

    expect(container).toMatchSnapshot();
  });
});
