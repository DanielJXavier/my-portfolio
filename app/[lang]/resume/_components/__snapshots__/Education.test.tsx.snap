// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education component renders the component 1`] = `
<div>
  <section>
    <h2
      class="text-[16pt] font-bold"
    >
      Education
    </h2>
    <ul
      class="flex flex-col gap-y-3"
    >
      <li
        class="items-baseline break-inside-avoid flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Master of Business Administration - MBA
            ,
             
            Strategic People Management
          </p>
          <p
            class="text-[12pt]"
          >
            Est√°cio
          </p>
        </div>
      </li>
      <li
        class="items-baseline break-inside-avoid flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt] font-bold"
          >
            Bachelor
            ,
             
            Computer Science
          </p>
          <p
            class="text-[12pt]"
          >
            Universidade FUMEC
          </p>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
